prune.go

```diff
@@ -100,6 +100,14 @@ func shouldPrune(oldCount, newCount int, pruneOptions PruneOptions) (bool, strin
 	return false, "transactions have not grown significantly"
 }
 
+func getOracle(db *mgo.Database, txns *mgo.Collection, txnsCount int, maxMemoryTxns int) (Oracle, func(), error) {
```

Comment:
Why does txnsCount get passed in? It can be easily obtained via `txns.Count()`. Ideally I assume you want completed txns here right, but the count of the txns collection will be pretty close.